addons/hr/models/hr_employee.py

```diff
@@ -278,8 +340,23 @@ def _cron_check_work_permit_validity(self):
 
     @api.model
     def get_view(self, view_id=None, view_type='form', **options):
+        # To be consistent with the global module behavior, we need
+        # to display the employee public view to the hr_coach group
+        # to make them able to see the full employee list
+        if self.env.user.has_group('hr.group_hr_coach') and \
+           not self.env.user.has_group('hr.group_hr_user') and \
+           view_type == 'form':
+
+            if options['action_id'] == self.env.ref('hr.open_view_employee_list_my').id:
+                return self.env['hr.employee.public'].get_view(view_id, view_type, **options)
+
+            if options['action_id'] == self.env.ref('hr.hr_employee_get_my_employee_action').id:
+                # force retrieve the private view
```

Comment:
The thing is, you are not just getting private employee view.. you were getting way more. That's why all the access errors